modules/nlu/src/backend/election/natural-election.ts

```diff
@@ -45,20 +48,47 @@ function electIntent(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstandin
   }
 }
 
-function extractElectedIntentSlot(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
-  if (!input.predictions) {
+export function extractElectedIntentSlot(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
+  if (!isNotNil(input.predictions)) {
     return input
   }
 
-  const elected = input.intent!
+  const elected = input.intent
+  if (!isNotNil(elected)) {
+    return input
+  }
 
-  if (!elected) {
+  const electedContext = input.predictions[elected.context]
+  if (!isContextPrediction(electedContext)) {
     return input
   }
 
-  const electedIntent = input.predictions[elected.context].intents.find(i => i.label === elected.name)
-  if (!electedIntent) {
+  const electedContextIntents = electedContext.intents
+  if (!Array.isArray(electedContextIntents)) {
```

Comment:
Can this really not be an array ? 

If so, we really have to fix the error at the source or adjust the typings

If not, I believe we really should trust the Typescript compiler and not do unnecessary checks

Reply:
The intents property is supposed to be an array of IntentPrediction-type (re-typed for this process as `EventUnderstanding`, unfortunately, but that's not the point), so checking that the value received is an array makes sense.

The problem this PR fixes **is** lack of type-checking. I think cutting corners is not a good idea, lest we leave behind any more bugs. I have tests covering all branches of this function so it should be rather solid now. Also, Typescript is powerful, but not powerful enough to hold any sway over runtime processes. Meaning when we run the program it's just Javascript. For this reason we should always typecheck even though the compiler is might be telling us everything is fine because at runtime in such a big application with processes talking to one another anything can replace the input.

For example, take this bit of code:
```typescript
function takeAString(x: string): string {
  return x;
}
takeAString((1 as any));
// big problem here, but it's possible and Typescript will let it slide
```

Also, we could potentially fix the error at the source, but I find that might be a related PR, or outside the direct scope of this one. The problem was literally in this function because of a missing null-check. (Or I should say hidden because of the bang operator.)

**NOTE:** __"...should trust the Typescript compiler and not do unnecessary checks..."__ no type check is unnecessary. I mean yes, but none in those I provided. I'm 100% confident about that.